### ISSUE TITLE
cnn_learner.get_preds() and validate does not apply  sigmoid()  as stated in book!

### ISSUE THREAD

**Issue by zhangjiekui** [fastai/fastbook#295]
![bug](https://user-images.githubusercontent.com/33198334/95821216-1c6d2780-0d5c-11eb-8e42-94eef772736c.png)
Then we can call the metric directly. Note that by default get_preds applies the output activation function (sigmoid, in this case) for us, so we'll need to tell accuracy_multi to not apply it:

in books [https://github.com/fastai/fastbook/blob/master/06_multicat.ipynb]

but it's not True! Just as bellow:

**Comment by zhangjiekui** [comment]
maybe should check "inp" in accuracy_multi() function:
def accuracy_multi(inp, targ, thresh=0.5, sigmoid=True):
    "Compute accuracy when `inp` and `targ` are the same size."
    inp,targ = flatten_check(inp,targ)
    if sigmoid: inp = inp.sigmoid()
    inp_flatted= inp.flatten()
    num_lt0=len(torch.nonzero(inp_flatted.lt(0.)))
    num_ge1=len(torch.nonzero(inp_flatted.gt(1.)))
    if num_lt0+num_ge1>0:
        print(inp[0])
        raise ValueError("‘Inputs'contain values less then 0 or greate then 1，so NO sigmoid() applied! ")
    return ((inp>thresh)==targ.bool()).float().mean()

**Comment by zhangjiekui** [comment]
Crazy! It‘s caused by the pass loss_func, but I can't understand it. Why?

![bigbug](https://user-images.githubusercontent.com/33198334/95866860-03d03200-0d9b-11eb-8dd3-c21623546aae.png)

**Comment by zhangjiekui** [comment]
The default loss_func is "fastai.losses.BCEWithLogitsLossFlat", which adding flat() and sigmoid() to the original "torch.nn.modules.loss.BCEWithLogitsLoss".

related to "https://github.com/fastai/fastbook/issues/172"
